Spigot-Server-Patches/0230-Reset-spawner-timer-when-spawner-event-is-cancelled.patch

```diff
@@ -0,0 +1,32 @@
+From 5e4e264244fe297e63d560b08ec49253c15d8692 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Mon, 31 Jul 2017 01:45:19 -0500
+Subject: [PATCH] Reset spawner timer when spawner event is cancelled
+
+
+diff --git a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
+index a5b5ef0f..cca09a16 100644
+--- a/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
++++ b/src/main/java/net/minecraft/server/MobSpawnerAbstract.java
+@@ -112,6 +112,9 @@ public abstract class MobSpawnerAbstract {
+                         {
+                             entity.fromMobSpawner = true;
+                         }
++
++                        flag = true; // Paper - moved up above cancellable event
++
+                         if (org.bukkit.craftbukkit.event.CraftEventFactory.callSpawnerSpawnEvent(entity, blockposition).isCancelled()) {
+                             continue;
+                         }
+@@ -121,8 +124,6 @@ public abstract class MobSpawnerAbstract {
+                         if (entityinsentient != null) {
+                             entityinsentient.doSpawnEffect();
+                         }
+-
+-                        flag = true;
```

Comment:
for a single line, the general convention is to comment it out and if needed leave a reason why it was commented out, makes tracking changes when we merge upstream later nicer as we can see and understand the modifications easily when git decides it wants to have a round with it :(